sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGenerateExec.scala

```diff
@@ -53,12 +53,54 @@ class GpuGenerateExecSparkPlanMeta(
   }
 
   override def convertToGpu(): GpuExec = {
-    GpuGenerateExec(
-      childExprs.head.convertToGpu().asInstanceOf[GpuGenerator],
-      gen.requiredChildOutput,
-      gen.outer,
-      gen.generatorOutput,
-      childPlans.head.convertIfNeeded())
+    // if child expression contains Stack, use GpuExpandExec instead
+    if (childExprs.exists(_.isInstanceOf[GpuStackMeta])) {
+      val stackMeta = childExprs.filter(_.isInstanceOf[GpuStackMeta]).head
+      val numRows = stackMeta.childExprs.head.convertToGpu()
```

Comment:
Lets use `extractLit` for this instead. It is going to be more generic that what we have here.

Reply:
Thanks, done.